_posts/2019-07-23-extracting-fsspec-from-dask.md

```diff
@@ -0,0 +1,133 @@
+file-system with the same interface, and finally a registry for all of these with various helper
+functions. Very little of this was specific to Dask, with only a couple of
+functions concerning themselves with building graphs and deferred execution. It did, however,
+raise the important issue that file-systems should be serializable and that there should
+be a way to specify a file to be opened, which is also serializable (and ideally supports
+transparent text and compression). 
+
+New File-Systems
+----------------
+
+I already mentioned the effort to make a local file-system class which met the same interface as
+the other ones which already existed. But there are more options that Dask users (and others) 
+might want, such as ssh, ftp, http, in-memory, and so on. Following requests from users to handle these options,
+we started to write more file-system interfaces, which all lived within `dask.bytes`; but it was unclear
+whether they should only support very minimal functionality, just enough to get something done from
+Dask, or a full set of file operations. 
+
+The in-memory file-system, in particular, existed in an extremely long-lived PR - it's not
+clear how useful such a thing is to Dask, when each worker has it's own memory, and so sees
+a different state of the "file-system". 
+
+Consolidation
+-------------
+
+[File-System Spec](https://github.com/intake/filesystem_spec), later `fsspec`, was born out of a desire
+to codify and consolidate the behaviours of the storage backends, reduce duplication, and provide the
+same functionality to all backends. In the process, it became much easier to write new implementation
+classes: see the [implementation](https://filesystem-spec.readthedocs.io/en/latest/api.html#built-in-implementations),
+which include interesting and highly experimental options such as the `CachingFileSystem`, which
+makes local copies of every remote read, for faster access the second time around. However, more
+important main-stream implementations also took shape, such as FTP, SSH, Memory and webHDFS
+(the latter being the best bet for accessing HDFS from outside the cluster, following all the
+problems building and authenticating with `hdfs3`).
+
+Furthermore, the new repo gave the opportunity to implement new features, which would then have
+further-reaching applicability than if they had been done in just selected repos. Examples include
+FUSE mounting, dictionary-style key-value views on file-systems
+(such as used by [zarr](https://zarr.readthedocs.io/en/stable/)), and transactional writing of
+files. All file-systems are serializable and pyarrow-compliant.
+
+Usefulness
+-----------
+
+Eventually it dawned on my that the operations offered by the file-system classes are very useful
+for people not using Dask too. Indeed, `s3fs`, for example, sees plenty of use stand-alone, or in
+conjunction with something like fastparquet, which can accept file-system functions to its method,
+or pandas.
```

Comment:
Ah, here we are.  I would send this up to the top.  I think that it's likely to be much more compelling than internal consistency.